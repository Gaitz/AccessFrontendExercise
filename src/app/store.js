import { configureStore } from "@reduxjs/toolkit";
import userListSlice from "../features/list/userListSlice";
import userDetailSlice from "../features/detail/userDetailSlice";

const store = configureStore({
  reducer: { list: userListSlice, detail: userDetailSlice },
});

export default store;
