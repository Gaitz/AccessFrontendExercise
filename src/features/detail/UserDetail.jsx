import React from "react";
import styles from "./UserDetail.module.css";
import { useParams } from "react-router-dom";

const UserDetail = () => {
  const { userLogin } = useParams();

  return (
    <>
      <h2 className={styles.pageTitle}>
        GitHub User: <em>{userLogin}</em> Detail
      </h2>
    </>
  );
};
export default UserDetail;
