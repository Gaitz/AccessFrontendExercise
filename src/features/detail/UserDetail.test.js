import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import store from "../../app/store";
import React from "react";
import UserDetail from "./UserDetail";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useParams: jest.fn().mockReturnValue("gaitz"),
}));

describe("UserDetail component test", () => {
  it("snapshot test", () => {
    const { container } = render(
      <Provider store={store}>
        <UserDetail></UserDetail>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
