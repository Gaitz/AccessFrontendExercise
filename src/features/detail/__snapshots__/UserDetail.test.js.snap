// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetail component test snapshot test 1`] = `
<div>
  <article
    class="userDetailComponent"
  >
    <h2
      class="pageTitle"
    >
      GitHub User: 
      <em />
       Detail
    </h2>
    <p
      class="pendingBlock"
    >
      Waiting for request 
    </p>
  </article>
</div>
`;
